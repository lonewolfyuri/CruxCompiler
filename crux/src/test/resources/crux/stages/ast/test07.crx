// test assignment

func main() : void {
  var x:bool;

  let x = not x and true;
}
