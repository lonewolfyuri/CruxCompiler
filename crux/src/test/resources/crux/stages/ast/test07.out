ast.DeclarationList(3,0)
  ast.FunctionDefinition(3,0)[Symbol(main), []]
    ast.StatementList(4,2)
      ast.VariableDeclaration(4,2)[Symbol(x)]
      ast.Assignment(6,2)
        ast.Name(6,6)[x]
        ast.OpExpr(6,16)[LOGIC_AND]
          ast.OpExpr(6,10)[LOGIC_NOT]
            ast.Dereference(6,14)
              ast.Name(6,14)[x]
          ast.LiteralBool(6,20)[TRUE]
